actors/evm/src/lib.rs

```diff
@@ -54,22 +54,23 @@ impl EvmContractActor {
             return Err(ActorError::illegal_argument("no bytecode provided".into()));
         }
 
-        // initialize contract state
-        let init_contract_state_cid =
-            Hamt::<_, U256, U256>::new(rt.store()).flush().map_err(|e| {
-                ActorError::unspecified(format!("failed to flush contract state: {e:?}"))
-            })?;
+        // create an empty storage HAMT to pass it down for execution.
+        let hamt = Hamt::<_, U256, U256>::new(rt.store().clone());
+
         // create an instance of the platform abstraction layer -- note: do we even need this?
-        let system = System::new(rt, init_contract_state_cid).map_err(|e| {
+        let mut system = System::new(rt, hamt).map_err(|e| {
             ActorError::unspecified(format!("failed to create execution abstraction layer: {e:?}"))
         })?;
```

Comment:
We now create a HAMT here with a cloned blockstore, and pass ownership down to the System. It's cleaner this way anyway.